### PR TITLE
Adding thread per request

### PR DESCRIPTION
**Pull request by alfredlucero**
TODO for team Mr. Robot Et Al: 
- please work on making our data structures related to the status handler thread-safe with locks i.e. mutexes
- in order to test, one must also alter the API to handle a partial request i.e. an indeterminate state if not all of the message is sent at once

### CODE REVIEW

**Comment by KyCodeHuynh** [comment]
Depending on what we hear back on Piazza, multithreading may be something we implement on our own repos.

**Comment by alfredlucero** [comment]
Yeah, that's why I stopped development on it as well until we clear it up but laid down future groundwork for any of us to pick up 👍. We'll continue to update each other once we figure it out.

**Comment by KyCodeHuynh** [comment]
It looks like we'll be [multithreading our own codebases](https://piazza.com/class/iwy6115h8ce37q?cid=86).

File: status_handler.cc [hunk 1]
```diff
@@ -18,11 +18,12 @@ const NginxConfig& config) {
...
 Response* response) {
-    // Get all the info we need from the server
-    const Server& s = *Server::GetInstance();
+    // Get all the info we need from the server and lock
+    Server& s = *Server::GetInstance();
+    s.Lock();
     const std::map<std::string, std::string>& hi = s.GetHandlersByUrl();
     const std::vector<std::pair<std::string, int> >& rh = s.GetRequestHistory();
-
+
```
Comment (gotitmemorize) [inline status_handler.cc]: random spaces 😈

**Comment by Wizzard033** [comment]
This is done now - the pull request just needs merged

File: server.h [hunk 2]
```diff
@@ -51,8 +52,8 @@ class Server  {
 public:
 
     // Returns the instance of the server using the singleton pattern
-    static std::unique_ptr<Server>& GetInstance() {
-        static std::unique_ptr<Server> instance;
+    static Server* GetInstance() {
+        static Server* instance = (Server*) malloc(sizeof(Server));
```
Comment (gotitmemorize) [inline server.h]: whenever status handler is called and GetInstance() is invoked, it mallocs something?

Comment (Wizzard033) [inline server.h]: Static initialization is only done once across all threads

**Pull request merged**